mmv1/products/cloudrunv2/api.yaml

```diff
@@ -973,51 +973,74 @@ objects:
+                          name: "httpHeaders"
+                          description: |-
+                            Custom headers to set in the request. HTTP allows repeated headers.
+                          item_type: !ruby/object:Api::Type::NestedObject
+                            properties:
+                              - !ruby/object:Api::Type::String
+                                name: name
+                                required: true
+                                description: |-
+                                  The header field name
+                              - !ruby/object:Api::Type::String
+                                name: value
+                                description: |-
+                                  The header field value
+                                default_value: ""
+                                send_empty_value: true
                     - !ruby/object:Api::Type::NestedObject
-                        name: "tcpSocket"
-                        description: |-
-                          TCPSocket specifies an action involving a TCP port. Exactly one of HTTPGet or TCPSocket must be specified.
-                        send_empty_value: true
-                        allow_empty_object: true
-                        # exactly_one_of:
-                        #   - template.0.containers.0.livenessProbe.0.httpGet
-                        #   - template.0.containers.0.livenessProbe.0.tcpSocket
-                        properties:
-                          - !ruby/object:Api::Type::Integer
-                            name: port
-                            description: |-
-                              Port number to access on the container. Must be in the range 1 to 65535. If not specified, defaults to 8080.
+                      name: "tcpSocket"
+                      description: |-
+                        TCPSocket specifies an action involving a TCP port. Exactly one of HTTPGet or TCPSocket must be specified.
+                      send_empty_value: true
+                      allow_empty_object: true
+                      # exactly_one_of:
+                      #   - template.0.containers.0.livenessProbe.0.httpGet
+                      #   - template.0.containers.0.livenessProbe.0.tcpSocket
+                      #   - template.0.containers.0.livenessProbe.0.grpc
+                      properties:
+                        - !ruby/object:Api::Type::Integer
+                          name: port
+                          description: |-
+                            Port number to access on the container. Must be in the range 1 to 65535. If not specified, defaults to 8080.
+                    - !ruby/object:Api::Type::NestedObject
+                      name: grpc
+                      description: |-
+                        GRPC specifies an action involving a GRPC port.
+                      send_empty_value: true
+                      allow_empty_object: true
```

Comment:
Why allow sending empty object?

Reply:
We try to avoid allow empty object since we are enabling features with empty blocks. From a terraform usage PoV it can be pretty confusing! Let me see what our teams position is on this... My immediate thought/preference is to make port required.

Usually apis have a separate field for feature enablement.
```hcl
resource "x" "y" {
  grpc {
    enabled = true
   }
}
```

Which allows the enablement of the feature to be more explicit. 

In my opinion an empty grpc block for feature enablement is slightly confusing to me with my background. 
```hcl
resource "x" "y" {
  grpc {
   }
}
```

let me see what my team says.

Reply:
response
```
Empty blocks should be avoided when possible. They are not well represented in Terraform.

`grpc{}` in understood by Terraform as []interface{}{nil}, not []interface{}{map[string]interface{}{}}
```

Reply:
It is better to honor the convention of the Terraform language syntax and prevent surprise to the provider users, so I will omit `send_empty_values: true` and `allow_empty_object: true` and mark `grpc.port` as required. 
In most cases, explicitly specifying a health check port number will not cause any notable inconvenience to Terraformers, and making infrastructure configurations explicit as code rather than implicit set will be usually preferred.